packages/flutter/lib/src/material/material_state.dart

```diff
@@ -365,6 +365,23 @@ abstract class MaterialStateBorderSide extends BorderSide implements MaterialSta
   ///
   /// The given callback parameter must return a non-null [BorderSide] in the
   /// default state.
+  ///
+  /// Usage:
+  /// ```dart
+  /// ChipTheme(
```

Comment:
This example would be a little clearer if getBorder was a closure - so that developers could see what it does.

Also: the doc for MaterialStateBorderSide should probably be updated to reflect the fact that resolveWith() is an option (you don't have to subclass anymore). In fact you might want to revise that example, since using resolveWith is simpler.